process_tree/process_tree.py

```diff
@@ -0,0 +1,558 @@
+        """Load recordings and generate visualizations."""
+        # Load all recordings
+        tree = self._load_recordings(recordings_dir)
+
+        # Generate outputs
+        # Generate SVG only if explicitly requested via output_svg parameter
+        if output_svg:
+            self.svg_renderer.render(tree, output_svg)
+
+        # Always show ASCII output
+        self.ascii_renderer.render(tree)
+
+    def _load_recordings(self, recordings_dir: Path) -> ProcessTree:
+        """Load all .undo files and build process tree."""
+        recording_files = list(recordings_dir.glob("*.undo"))
+        if not recording_files:
+            raise ValueError(f"No .undo files found in {recordings_dir}")
+
+        print(f"Found {len(recording_files)} recording files")
+
+        tree = ProcessTree()
+        for recording_file in recording_files:
+            pid, ppid, start_time = self.parser.extract_process_info(recording_file)
+            if pid is not None:
+                process = Process(pid, ppid, str(recording_file), start_time)
+                tree.add_process(process)
+                print(
+                    f"Loaded: PID {pid}, PPID {ppid}, Start: {start_time:.9f}, "
+                    f"File: {recording_file.name}"
+                )
+
+        tree.build_relationships()
+        return tree
+
+
+if HAS_GDB:
+
+    class ProcessTreeCommand(gdb.Command):
+        """
+        Visualize process trees from .undo recording files.
+
+        Usage: process-tree RECORDINGS_DIR [--output-svg FILE]
+
+        Arguments:
+            RECORDINGS_DIR: Directory containing .undo recording files
+
+        Options:
+            --output-svg FILE: Output SVG file path (generates SVG in addition to ASCII)
+
+        By default, only ASCII tree output is shown. Use --output-svg to also generate an SVG.
+
+        Examples:
+            process-tree /path/to/recordings
+            process-tree /path/to/recordings --output-svg tree.svg
+        """
+
+        def __init__(self) -> None:
+            super().__init__("process-tree", gdb.COMMAND_USER, gdb.COMPLETE_FILENAME)
+
+        def invoke(self, argument: str, _from_tty: bool) -> None:
+            """Execute the process-tree command."""
+            if not argument:
+                raise gdb.GdbError("Usage: process-tree RECORDINGS_DIR [--output-svg FILE]")
+
+            # Parse arguments
+            args = shlex.split(argument)
+            recordings_dir = Path(args[0]).expanduser()
+            output_svg = None
+
+            # Parse optional arguments
+            i = 1
+            while i < len(args):
+                if args[i] == "--output-svg":
+                    if i + 1 >= len(args):
+                        raise gdb.GdbError("--output-svg requires a filename")
+                    output_svg = str(Path(args[i + 1]).expanduser())
+                    i += 2
+                else:
+                    raise gdb.GdbError(f"Unknown argument: {args[i]}")
```

Comment:
Not the nicest way to parse arguments, but our own arugment parsing is not in the public API, so this is fine.